cider/exec/template/ArithmeticIR.cpp

```diff
@@ -850,6 +917,105 @@ llvm::Function* CodeGenerator::getArithWithOverflowIntrinsic(
   return llvm::Intrinsic::getDeclaration(cgen_state_->module_, fn_id, type);
 }
 
+llvm::Value* CodeGenerator::codegenArithWithDivZeroCheckForArrow(
+    const Analyzer::BinOper* bin_oper,
+    FixedSizeColValues* lhs,
+    FixedSizeColValues* rhs,
+    const std::string& null_check_suffix,
+    const SQLTypeInfo& ti) {
+  AUTOMATIC_IR_METADATA(cgen_state_);
+  auto lhs_lv = lhs->getValue(), rhs_lv = rhs->getValue();
+  auto lhs_null = lhs->getNull(), rhs_null = rhs->getNull();
+  llvm::BasicBlock* div_ok{nullptr};
+  llvm::BasicBlock* div_zero{nullptr};
+  div_ok = llvm::BasicBlock::Create(
+      cgen_state_->context_, "div_ok", cgen_state_->current_func_);
+  if (!null_check_suffix.empty()) {
+    codegenSkipOverflowCheckForNullForArrow(lhs_null, rhs_null, div_ok, ti);
```

Comment:
better name codegenSkipCheckForNullForArrow?

Reply:
This name is changed based on origin method name and the method only contains overflow check inside which makes this name more suitable.